### ISSUE TITLE
Rename *_mut to *_inplace and change return type

### ISSUE THREAD

**Issue by jturner314** [termoshtt/ndarray-linalg#83]
Methods that modify the input array in-place, such as `solve_mut` and `cholesky_mut`, would be better named `*_inplace`, such as `solve_inplace` and `cholesky_inplace`, to maintain consistency with `ndarray`. (See `ArrayBase::map_inplace` and `ArrayBase::mapv_inplace`.) Additionally, it would be clearer for those methods not to return a reference to the array that is being modified. In other words, I suggest changing:
```rust
fn solve_mut<'a, Sb: DataMut<Elem = A>>(&self, b: &'a mut ArrayBase<Sb, Ix1>) -> Result<&'a mut ArrayBase<Sb, Ix1>>;
```
to
```rust
fn solve_inplace<'a, Sb: DataMut<Elem = A>>(&self, b: &'a mut ArrayBase<Sb, Ix1>) -> Result<()>;
```
and
```rust
fn cholesky_mut(&mut self, uplo: UPLO) -> Result<&mut Self>;
```
to
```rust
fn cholesky_inplace(&mut self, uplo: UPLO) -> Result<()>;
```
etc.

What do you think? I can submit a PR with this change.

**Comment by termoshtt** [comment]
> Methods that modify the input array in-place, such as solve_mut and cholesky_mut, would be better named *_inplace, such as solve_inplace and cholesky_inplace, to maintain consistency with ndarray

Good suggestion. `*_inplace` is appropriate.

> Additionally, it would be clearer for those methods not to return a reference to the array that is being modified.

The returned reference is useful in several circumstance. First, it enable us to chain the methods:

```rust
a1.solve_mut(&mut b)?.norm();
a2.cholesky_mut(UPLO::Upper)?.dot(&b);
```

Second, we can rename the result reference naturally without generating new reference.

```rust
let lower = a.cholesky_mut(UPLO::Upper)?;
```

You may think I should use `*_into` method for these case, however, `*_into` method needs to consume the reference and it is sometimes prohibited or we need to use `std::mem::replace`, e.g. to rewrite a member of some struct.

**Comment by jturner314** [comment]
Okay, that makes sense. I just pushed a PR with the renaming.

**Comment by jturner314** [comment]
Fixed by #84.
